hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/listener/ClientSmartListenerService.java

```diff
@@ -200,6 +202,41 @@ public void run() {
         }, 1, 1, TimeUnit.SECONDS);
     }
 
+    private void openConnectionsToMembers() {
+        ClientClusterService clientClusterService = client.getClientClusterService();
+        Collection<Member> memberList = clientClusterService.getMemberList();
+        long startTimeMillis = System.currentTimeMillis();
+        while (true) {
+            Exception exception = null;
+            for (Member member : memberList) {
+                try {
+                    clientConnectionManager.getOrConnect(member.getAddress());
+                } catch (Exception e) {
+                    if (!clientClusterService.getMemberList().contains(member)) {
+                        //member left the cluster no need to connect
+                        continue;
+                    }
+                    exception = e;
+                }
+            }
+            if (exception == null) {
+                return;
+            }
+            ClientInvocationServiceImpl invocationService = (ClientInvocationServiceImpl) client.getInvocationService();
+            long invocationTimeoutMillis = invocationService.getInvocationTimeoutMillis();
+            long currentTimeMillis = System.currentTimeMillis();
+            long elapsedTime = currentTimeMillis - startTimeMillis;
+            if (elapsedTime > invocationTimeoutMillis) {
+                throw new OperationTimeoutException("Adding listeners are timed out."
+                        + " Current Time: " + currentTimeMillis + " ms ,"
+                        + " Start Time : " + startTimeMillis + " ms, "
+                        + " Elapsed time : " + elapsedTime + " ms. ", exception);
```

Comment:
can you also add the invocationTimeoutMillis which was used as a reference?

Reply:
I have added the invocation timeout millis